app/tasks/maintenance_tasks/task.rb

```diff
@@ -16,6 +16,9 @@ class NotFoundError < NameError; end
     # @api private
     class_attribute :throttle_conditions, default: []
 
+    class_attribute :collection_builder_strategy,
+      default: NullCollectionBuilder.new
```

Comment:
I think we may want a different instance per Task instance instead of one per class. That way we could handle memoization of `collection` when we want to provide an implicit `count`.

Also maybe the strategy should take the task instance when it's being instantiated, instead of for each method?